internal/backends/sqlite/metadata/registry.go

```diff
@@ -179,7 +235,10 @@ func (r *Registry) CollectionDrop(ctx context.Context, dbName string, collection
 		return false, nil
 	}
 
-	tableName := r.CollectionToTable(collectionName)
+	tableName, err := r.CollectionGet(ctx, dbName, collectionName)
+	if err != nil {
+		return false, nil
```

Comment:
I think we should return `return false, nil` only case where collection doesn't exist. Other case related to DB should return lazy error.